auth/common.cc

```diff
@@ -190,7 +191,11 @@ future<> announce_mutations(
         std::vector<data_value_or_unset> values,
         seastar::abort_source* as,
         std::optional<::service::raft_timeout> timeout) {
-    auto group0_guard = co_await group0_client.start_operation(as, timeout);
+    utils::composite_abort_source composite_as;
+    if (as) {
+        composite_as.add(*as);
+    }
```

Comment:
Please dig deeper (or rather, go opposite of deeper? up the callstack) to investigate further if this can be avoided.
If not, we need to leave a FIXME and open an issue (but preferably should be fixed right here.)

By masking the problem like this we can potentially shoot ourselves in our feet, with another shutdown hang in the future, if this `announce_mutations` call is started concurrently.

Reply:
Done ... I was stopping here because the change ripples through even into the `add_entry()`, but it looks that it can be changed too and it looks that all callers use an existing object (so we can use a reference).

The only thing is that it makes the refactoring relatively large - but it is kept in a separate commit.